ngraph/core/reference/include/ngraph/runtime/reference/proposal.hpp

```diff
@@ -0,0 +1,445 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/proposal.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace details
+            {
+                template <typename T>
+                struct ProposalBox
+                {
+                    T x0;
+                    T y0;
+                    T x1;
+                    T y1;
+                    T score;
+                };
+
+                static std::vector<float> generate_anchors(const op::ProposalAttrs& attrs,
+                                                           std::vector<float>& anchors)
+                {
+                    // Framework specific parameters
+                    auto coordinates_offset = attrs.framework == "tensorflow" ? 0.0f : 1.0f;
+                    auto round_ratios = attrs.framework == "tensorflow" ? false : true;
```

Comment:
Consider `const bool round_rations = attrs.framework != "tensorflow";`

Reply:
I gave it a moment and I don't see any benefits other than the lines being a bit shorter, as I responded to your previous comment, my decision is to leave it as it is.